HIP/hip-850.md

```diff
@@ -0,0 +1,69 @@
+---
+hip: 850
+title: Enhancing Supply Key Functionality for NFT Updates in Treasury Wallet
+author: Patches (@HGraphPunks)
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Review
+created: 2024-01-02
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/660
+requested-by: HGraphPunks
+updated: 2024-01-24
+---
+
+## Abstract
+
+This HIP proposes enhancing the Supply Key functionality within the Hedera ecosystem, specifically enabling the Supply Key to update an NFT metadata field while the NFT is held in the treasury wallet. This ensures secure and controlled adaptability of NFTs, meeting dynamic needs while maintaining their integrity post-distribution.
+
+## Motivation
+
+Currently, there is a limitation in dynamically updating NFTs in a controlled manner. The proposed enhancement to the Supply Key functionality addresses this gap, allowing for secure and selective updates of NFTs within the treasury wallet, a feature particularly relevant for various evolving use cases in the Hedera ecosystem.
+
+## Rationale
+
+The enhancement focuses on granting the Supply Key the capability to update the data of an NFT serial number exclusively while it is in the treasury wallet. This approach ensures security and control, preventing unauthorized modifications once the NFT is distributed.
+
+## User Stories
+
+- As a game developer, I want to update game piece NFTs with player stats during gameplay, ensuring the data becomes immutable once the player regains ownership.
+- As an event organizer, I need to update event ticket NFTs with attendee information while they are in the treasury wallet, guaranteeing data immutability post-transfer.
+
+## Specification
+
+This HIP requires an update to the Supply Key mechanism, enabling it to modify the metadata of NFTs while they are held in the treasury wallet. The NFT data should remain immutable once distributed, ensuring integrity and trust in the NFT ecosystem.
+
+## Backwards Compatibility
+
+This proposal introduces no backward incompatibilities within the current Hedera ecosystem.
+
+## Security Implications
+
+The primary security consideration is the prevention of unauthorized data modifications to NFTs post-distribution. The treasury wallet requirement for updates addresses this concern effectively. 
+
+There is no added security concern with the ability of Supply Key to adapt the metadata when in the treasury wallet as the Supply Key can also burn and mint new tokens. The security implications of sending a NFT to a treasury wallet that has a Supply Key enabled does not become worse or better with this HIP.
```

Comment:
I think there's an application level security concern here.
For NFTs that are used as authentication for access to experiences or other items the ability to modify the data means there's impacts on the number of NFTs that comply or don't comply with the NFT collection intent.
For example in the game user story example  if a collection of 100 promised that only 5 NFTs existed that had special traits, then the ability of the treasury to change that would change the impression and understanding of the collection overall and the ecosystems built on it

Reply:
The burn case is fine as that NFT serial is gone a different one may be issued same or different.
The issue is that the same NFT serial can morph unexpectedly across the collection and change assumptions and inferences on the collection.
Not against it just exploring conceptual management by users.

Reply:
I think this should be called out explicitly in the Backwards compatibility section. For some there is a change in define so we should call it out as a conceptual difference for those that don't expect supply_key to be able to to update metadata of a single NFT after mint